File: src/dataset/typescript/space-age_Exercism-typescript-shybyte/openAI/ChatGPT4_space-age_v3.ts
const EARTH_YEAR_IN_SECONDS = 31557600;

export default class SpaceAge {
    public seconds: number;

    private static readonly PLANET_YEAR_LENGTHS = {
        Earth: 1.0,
        Mercury: 0.2408467,
        Venus: 0.61519726,
        Mars: 1.8808158,
        Jupiter: 11.862615,
        Saturn: 29.447498,
        Uranus: 84.016846,
        Neptune: 164.79132,
    };

    constructor(seconds: number) {
        this.seconds = seconds;
    }

    onEarth() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Earth);
    }

    onMercury() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Mercury);
    }

    onVenus() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Venus);
    }

    onMars() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Mars);
    }

    onJupiter() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Jupiter);
    }

    onSaturn() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Saturn);
    }

    onUranus() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Uranus);
    }

    onNeptune() {
        return this.calculateAge(SpaceAge.PLANET_YEAR_LENGTHS.Neptune);
    }

    private calculateAge(earthYears: number) {
        return Math.round((this.seconds / (earthYears * EARTH_YEAR_IN_SECONDS)) * 100) / 100;
    }
}

File: src/dataset/Java/space-age_exercism-java-ThomasZumsteg/openAI/ChatGPT4_space-age_v3.java
public class SpaceAge {
    private final double seconds;
    private static final double EARTH_YEAR_IN_SECONDS = 31557600.0;

    public SpaceAge(Long seconds) {
        this.seconds = seconds;
    }

    public SpaceAge(Integer seconds) {
        this.seconds = seconds;
    }

    public Long getSeconds() {
        return (long) seconds;
    }

    public double onEarth() {
        return seconds / EARTH_YEAR_IN_SECONDS;
    }

    public double onMercury() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 0.2408467);
    }

    public double onVenus() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 0.61519726);
    }

    public double onMars() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 1.8808158);
    }

    public double onJupiter() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 11.862615);
    }

    public double onSaturn() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 29.447498);
    }

    public double onUranus() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 84.016846);
    }

    public double onNeptune() {
        return seconds / (EARTH_YEAR_IN_SECONDS * 164.79132);
    }
}